nixos/modules/programs/dconf.nix

```diff
@@ -4,13 +4,24 @@ with lib;
 
 let
   cfg = config.programs.dconf;
-
-  mkDconfProfile = name: path:
-    {
-      name = "dconf/profile/${name}";
-      value.source = path; 
-    };
-
+  cfgDir = pkgs.symlinkJoin {
+    name = "dconf-system-config";
+    paths = map (x: "${x}/etc/dconf") cfg.packages;
+    postBuild = ''
+      mkdir -p $out/profile
+      mkdir -p $out/db
+    '' + (
+      concatStringsSep "\n" (
+        mapAttrsToList (
+          name: path: ''
+            ln -s ${path} $out/profile/${name}
+          ''
+        ) cfg.profiles
+      )
```

Comment:
Why won't the profiles already be linked by `symlinkJoin`?

Reply:
I was also confused by this but profiles are not packages, they are just file paths.

Reply:
yes, the symlinkJoin is only for the option `packages` and this look is for the option `profiles`. So I think this is correct.

Reply:
I understand, thanks!